src/codegen/traverse.lisp

```diff
@@ -7,11 +7,19 @@
    (#:util #:coalton-impl/util))
   (:export
    #:action
+   #:count-applications
+   #:count-nodes
+   #:*traverse*
    #:traverse
    #:traverse-with-binding-list))
 
 (in-package #:coalton-impl/codegen/traverse)
 
+(defvar *traverse* nil
+  "The recursive function reference for the current traversal. This will
+be bound for the dynamic extent of any traversal. Inside of an
+`action`, `(funcall *traverse* node)` may be used to invoke recursion.")
```

Comment:
Most changes in this file are renaming and documentation updates to reflect this